### ISSUE TITLE
[Bug]:  Basingstoke and Deane is broken since the 8th of Feb

### ISSUE THREAD

**Issue by AndreaCCIE** [mampfes/hacs_waste_collection_schedule#1887]
### I Have A Problem With:

A specific source

### What's Your Problem

The service no longer downloads the waste updates. I tried 1.46 and the master.

### Logs

```Shell
Logger: waste_collection_schedule.source_shell
Source: custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py:136
integration: waste_collection_schedule (documentation)
First occurred: 11:26:41 (1 occurrences)
Last logged: 11:26:41

fetch failed for source Basingstoke and Deane Borough Council: Traceback (most recent call last): File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch entries = self._source.fetch() ^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/basingstoke_gov_uk.py", line 65, in fetch date=datetime.strptime(date_str, "%A, %d %B %Y").date(), ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/_strptime.py", line 554, in _strptime_datetime tt, fraction, gmtoff_fraction = _strptime(data_string, format) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.12/_strptime.py", line 333, in _strptime raise ValueError("time data %r does not match format %r" % ValueError: time data 'none / unknown' does not match format '%A, %d %B %Y'
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: basingstoke_gov_uk
      args:
        uprn: "1000809XXXX"
      customize:
        - type: Garden
          show: True
        - type: Waste
          show: True
        - type: Recycling
          show: True
        - type: Glass
          show: True
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
Our tests UPRNs (`100060234732`, `100060218986`, `100060235836`, `100060224194`) seem to work fine. Could you please enter your address in https://www.basingstoke.gov.uk/bincollections and check weather collection dates are actually shown. If not, could you please provide an HTML snippet starting with the `Waste collection dates` Headline or at least a screenshot of the section where you see `none / unknown` instead of the actual date.

**Comment by 5ila5** [comment]
After the fix in #1888 the source should still work for the collections that do have a date and just log a warning for the none / unknown dates. You could try your neighbors addresses, maybe that works.

This will be part of release 1.47.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)

**Comment by AndreaCCIE** [comment]
Thanks, I can see the error is now managed properly

![Screenshot 2024-03-10 at 13 53 07](https://github.com/mampfes/hacs_waste_collection_schedule/assets/48698293/2fb19172-c6a0-4687-8d45-ac0a7ad74dbd)
